packages/react/src/components/Flex/Flex.tsx

```diff
@@ -90,20 +90,27 @@ class Flex extends UIComponent<ReactProps<FlexProps>> {
   renderChildren = (gapClasses: string) => {
     const { column, gap, children } = this.props
 
-    return React.Children.map(children, (child: React.ReactElement<any>, index) => {
-      const childElement =
-        child.type && ((child.type as any) as typeof FlexItem).__isFlexItem
+    let isFirstElement = true
+    return React.Children.map(children, (child: any) => {
+      const maybeChildElement =
+        child && child.type && ((child.type as any) as typeof FlexItem).__isFlexItem
           ? React.cloneElement(child, {
               flexDirection: column ? 'column' : 'row',
             })
           : child
 
-      const renderGap = index !== 0
+      const renderGap = !isFirstElement
+      if (maybeChildElement) {
+        isFirstElement = false
+      }
+
       return (
-        <>
-          {renderGap && gap && <Flex.Gap className={cx(`${Flex.className}__gap`, gapClasses)} />}
-          {childElement}
-        </>
+        maybeChildElement && (
```

Comment:
Do we need a condition there?
Without it we will get:
```jsx
<>
  {null}
  {null}
</>
```
But it will not affect the markup

Reply:
Nevermind, 

> If children is null or undefined, this method will return null or undefined rather than an array.
https://reactjs.org/docs/react-api.html#reactchildrenmap

Reply:
yes - but, essentially, we won't like to draw anything if falsy value is provided - and this is exactly the effect logic provides. Thus I would rather leave it the way it is now - as a benefit this won't introduce any additional React elements to output elements tree